lfs/transfer_queue.go

```diff
@@ -231,7 +276,12 @@ func (q *TransferQueue) transferWorker() {
 		}
 
 		if err := transfer.Transfer(cb); err != nil {
-			q.errorc <- err
+			if q.canRetry(err) {
```

Comment:
I might push for a wrapper function too, especially considering this is the only place that uses `retry()` and `canRetry()`. However, the `if` statements are different enough that they would have to accept a function callback or something. 

```go
q.queueRetry(err, func() {
  tracerx.Printf("tq: retrying object %s", transfer.Oid())
  q.retry(transfer)
})
```

That may save you some lines of code, but I think the current form is easier to read.